sup3r/preprocessing/samplers/cc.py

```diff
@@ -0,0 +1,179 @@
+        assert lr == data[0] and hr == data[1], msg
+        if t_enhance == 1:
+            hr = data.daily
+        if s_enhance > 1:
+            lr = lr.coarsen(
+                {
+                    Dimension.SOUTH_NORTH: s_enhance,
+                    Dimension.WEST_EAST: s_enhance,
+                }
+            ).mean()
+        data = Sup3rDataset(low_res=lr, high_res=hr)
+        super().__init__(
+            data=data,
+            sample_shape=sample_shape,
+            batch_size=batch_size,
+            t_enhance=t_enhance,
+            s_enhance=s_enhance,
+            feature_sets=feature_sets,
+        )
+
+    def check_for_consistent_shapes(self):
+        """Make sure container shapes are compatible with enhancement
+        factors."""
+        enhanced_shape = (
+            self.lr_data.shape[0] * self.s_enhance,
+            self.lr_data.shape[1] * self.s_enhance,
+            self.lr_data.shape[2] * (1 if self.t_enhance == 1 else 24),
+        )
+        msg = (
+            f'hr_data.shape {self.hr_data.shape} and enhanced '
+            f'lr_data.shape {enhanced_shape} are not compatible with '
+            f'the given enhancement factors t_enhance = {self.t_enhance}, '
+            f's_enhance = {self.s_enhance}'
+        )
+        assert self.hr_data.shape[:3] == enhanced_shape, msg
+
+    def reduce_high_res_sub_daily(self, high_res, csr_ind=0):
+        """Take an hourly high-res observation and reduce the temporal axis
+        down to lr_sample_shape[2] * t_enhance time steps, using only daylight
+        hours on the middle part of the high res data.
+
+        Parameters
+        ----------
+        high_res : T_Array
+            5D array with dimensions (n_obs, spatial_1, spatial_2, temporal,
+            n_features) where temporal >= 24 (set by the data handler).
+        csr_ind : int
+            Feature index of clearsky_ratio. e.g. self.data[..., csr_ind] ->
+            cs_ratio
+
+        Returns
+        -------
+        high_res : T_Array
+            5D array with dimensions (n_obs, spatial_1, spatial_2, temporal,
+            n_features) where temporal has been reduced down to the integer
+            lr_sample_shape[2] * t_enhance. For example if hr_sample_shape[2]
+            is 9 and t_enhance = 8, 72 hourly time steps will be reduced to 9
+            using the center daylight 9 hours from the second day.
+
+        Note
+        ----
+        This only does something when `1 < t_enhance < 24.` If t_enhance = 24
+        there is no need for reduction since every daily time step will have 24
+        hourly time steps in the high_res batch data. Of course, if t_enhance =
+        1, we are running for a spatial only model so this routine is
+        unnecessary.
+
+        *Needs review from @grantbuster
+        """
+        if self.t_enhance not in (24, 1):
+            high_res = self.get_middle(high_res, self.hr_sample_shape)
+            high_res = nsrdb_reduce_daily_data(
```

Comment:
minor but in these methods (two places) why did you move np.round -> np.ceil?

Reply:
So I could account for an odd index for a mid point

Reply:
wouldn't round do this? This is maybe a moot point because the high res shape should always be a multiple of 24.

Reply:
Yeah maybe round would do this also. High res doesn't need to be a multiple of 24 though. see https://github.com/NREL/sup3r/blob/a6acb905c310c75908dd899e8725203791e0c115/tests/batch_handlers/test_bh_h5_cc.py#L31. I guess before it goes through reduce daily it is always a multiple of 24 though.